src/api/helpers/formatEntityData.ts

```diff
@@ -99,6 +99,82 @@ export function getWorkBasicInfo(work: Record<string, unknown> | null | undefine
 		};
 }
 
+/**
+ * A function to extract structured author credit details from an Edition ORM entity.
+ * @function
+ * @param {object} edition - an edition ORM entity
+ * @returns {object|null} an object containing the author credit data, or null if not available.
+ *
+ * @example
+ * 		const edition = await orm.func.entity.getEntity(orm, 'Edition', bbid, relations);
+ *		filterAuthorCredit(edition);
+ *		/* => {
+ *			"authorCount": 1,
+ *			"id": 135,
+ *			"names": [
+ *				{
+ *					"authorBBID": "e5c4e68b-bfce-4c77-9ca2-0f0a2d4d09f0",
+ *					"authorCreditID": 135,
+ *					"name": "J. K. Rowling"
+ *				}
+ *			]
+ *		}
+ */
+
+export function filterAuthorCredit(edition: any) {
+	if (_.isNil(edition) || _.isNil(edition.authorCredit)) { return null; }
+
+	const credit = edition.authorCredit;
+	return {
+		authorCount: credit.authorCount || null,
+		id: credit.id || null,
+		names: (credit.names || []).map(({authorCreditID, authorBBID, name}) => ({
+			authorBBID,
+			authorCreditID,
+			name
+		}))
+	};
+}
+
+/**
+ * A function to extract basic publisher information from an Edition ORM entity.
+ * @function
+ * @param {object} edition - an edition ORM entity
+ * @returns {object|null} an object containing publisher set data, or null if not available.
+ *
+ * @example
+ * 		const edition = await orm.func.entity.getEntity(orm, 'Edition', bbid, relations);
+ *		filterPublisherSet(edition);
+ *		/* => {
+ *			"id": 2217,
+ *			"publishers": [
+ *				{
+ *					"bbid": "d30d7df9-eb6f-4bd5-a69f-a9a3362a6f4a",
+ *					"ended": false,
+ *					"name": "Bloomsbury",
+ *					"publisherType": "Publisher",
+ *					"sortName": "Bloomsbury"
+ *				}
+ *			]
+ *		}
+ */
+
+export function filterPublisherSet(edition: any) {
+	if (_.isNil(edition) || _.isNil(edition.authorCredit)) { return null; }
+
+	const {publisherSet} = edition;
+	return {
+		id: publisherSet.id || null,
+		publishers: (publisherSet.publishers || []).map(({bbid, name, sortName, ended, publisherType}) => ({
```

Comment:
Same question here as above, wondering if the mapping function does anything.

Reply:
Similarly, here I am just sharing the properties which I feel might be useful. 
Here is a list of all the available properties:
bbid, dataid, revisionId, master, annotationId, disambiguationId, disambiguation, defaultAliasId, name, sortName, beginYear, beginMonth, beginDay, endYear, endMonth, endDay, ended, areaId, typeId, publisherType, aliasSetId, identifierSetId, relationshipSetId, type, _pivot_set_id, _pivot_publisher_bbid, beginDate, endDate